edx_exams/apps/api/v1/views.py

```diff
@@ -1 +1,287 @@
 # Create your views here.
+from random import randint
+from urllib.parse import urljoin
+
+from django.conf import settings
+from django.http import JsonResponse
+from django.shortcuts import redirect, render
+from django.urls import reverse
+from django.utils.crypto import get_random_string
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+
+from lti_consumer.lti_1p3.exceptions import (
+    BadJwtSignature,
+    InvalidClaimValue,
+    MalformedJwtToken,
+    MissingRequiredClaim,
+    NoSuitableKeys,
+    TokenSignatureExpired,
+    UnauthorizedToken,
+)
+
+from edx_exams.apps.api.api import (
+    get_lti1p3_consumer,
+    get_lti_preflight_url,
+    get_resource_link,
+    get_optional_user_identity_claims,
+)
+
+def lti_start_proctoring(request):
+    """
+    This is the view that'll redirect the learner to the Tool's initial OIDC login initiation URL.
+
+    The Assessment Platform needs to know the Tool's OIDC login initiation URL.
+    This information is exchanged out-of-band during the registration phase.
+    """
+    # TODO: Here we'd do all the start of proctoring things
+
+    lti_message_hint = 'LtiStartProctoring'
+    preflight_url = get_lti_preflight_url(lti_message_hint)
+
+    return redirect(preflight_url)
+
+def lti_end_assessment(request):
+    """
+    This is the view that'll redirect the learner to the Tool's initial OIDC login initiation URL.
```

Comment:
comment copy/pasted?

Reply:
Because the endAssessment message is a new Assessment Platform originating message, it has to go through the entire OpenID Connect LTI launch flow all over again in order to send the endAssessment message, so this is technically still a correct comment. I can clarify the comment with that detail. Is this something else unclear about the comment?

Reply:
so the first part of ending is to login? That is super confusing but "has to go through the entire OpenID Connect LTI launch flow all over again in order to send the endAssessment message" would clear up why we're going back to the same place

Reply:
"Login" is kind of confusing here. More specifically, calling this endpoint initiates the OpenID Connect authorization flow, which is a necessary step when sending any LTI related message, because the Assessment Platform acts as the identity provider here. The IMS calls the endpoint that does this the "3rd Party initiated login end-point". The rationale gets more into the OIDC specification, which the IMS pulls heavily from, which I'm not as familiar with. I'll see if I can make this clearer.